### PR TITLE
gdal: use pcre2

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gdal.rb [hunk 1]
```diff
@@ -92,6 +92,7 @@ def install
       "--with-pg=yes",
       "--with-png=#{Formula["libpng"].opt_prefix}",
       "--with-spatialite=#{Formula["libspatialite"].opt_prefix}",
+      "--with-pcre2=yes",
```
Comment (carlocab) [inline Formula/gdal.rb:95]: No `opt_prefix`?

**Review by carlocab — COMMENTED** [review]
`revision` bump?

File: Formula/gdal.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/gdal.rb:95]: `brew linkage` was already happy with the current change. I'm not sure if this takes such an argument

**Comment by SMillerDev** [comment]
It's not important to users so I'd leave the rebuild to the next release.

**Comment by carlocab** [comment]
> It's not important to users so I'd leave the rebuild to the next release.

I dunno, I'm worried about us starting to build things against `gdal` + `pcre2` with users installing them against `gdal` + `pcre`. It may not be an issue, but it might also be.

**Review by carlocab — APPROVED** [review]
Thanks @SMillerDev ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**